### ISSUE TITLE
array-bracket-spacing false-positive when accessing array index in array.

### ISSUE THREAD

**Issue by docksteaderluke** [eslint/eslint#4083]
I have the following in my .eslintrc file:
```json
"array-bracket-spacing" : [ 1, "always", {
  "objectsInArrays": false,
  "arraysInArrays": false
}]
```
...and the following code:
```js
var theBestApples = [ apples[i], apples[j] ]
```
... and the following ```array-bracket-spacing``` warning from eslint: ```There should be no space before ']' at line...```

My guess is that eslint is seeing ```[j]``` and incorrectly recognizing it as an array literal rather than reference of index ```j``` on the ```apples``` array. The warning only gets triggered when the array reference is the last element in the array. Eliminating the space before the final ```]``` removes the warning.

I am using eslint version ```1.1.0```.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by docksteaderluke** [comment]
I'm using eslint version ```1.1.0```

**Comment by nzakas** [comment]
Can you try the latest version of ESLint?

**Comment by docksteaderluke** [comment]
Updated to v1.6.0. No change. Still warns incorrectly.

**Comment by nzakas** [comment]
OK thanks, we will take a look.

**Comment by alberto** [comment]
Working on this

**Comment by alberto** [comment]
It also fails for other cases:
- nested array patterns
```
{ code: "var [ x, [ y, z ]] = arr;", ecmaFeatures: { destructuring: true }, options: ["always", {"arraysInArrays": false}] },
```
- function definitions
```
{ code: "var foo = [ function(){} ];", options: ["always", {objectsInArrays: false}] },
```
Maybe the issue should be renamed to reflect that.
